### ISSUE TITLE
fix(no-bare-urls): false positive for URLs inside HTML attribute values

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#150]
## Overview

The \`no-bare-urls\` rule incorrectly flags URLs that appear inside HTML attribute values (e.g. \`href\`, \`src\`) as bare URLs.

## Problem

When an HTML tag is used in a Markdown file, URLs inside attributes are reported as violations even though they are not bare URLs — they are properly embedded in HTML markup.

**Example:**

\`\`\`html
<a href="https://example.com"><img src="https://example.com/image.gif" width="800" alt="Demo"></a>
\`\`\`

**Current output:**
\`\`\`
README.md:11: [error] no-bare-urls: bare URL found, use angle brackets or a Markdown link: https://example.com
README.md:11: [error] no-bare-urls: bare URL found, use angle brackets or a Markdown link: https://example.com/image.gif
\`\`\`

**Expected output:** No violations — URLs inside HTML attributes should not be treated as bare URLs.

## Proposed Fix

Update the \`no-bare-urls\` detection logic to skip URLs that appear inside HTML attribute values (i.e. within quoted attribute strings like \`href="..."\` or \`src="..."\`).

## References

- Discovered in PR #149 where an HTML \`<img>\` tag was used to control GIF display size in README.md

**Comment by fekete-robert** [comment]
You also get the same false positive if the bare url is inside an html comment, like

```markdown
Some text

<!-- FIXME update when issue is fixed https://example.com/ -->
```
